src/bin/common/solving_loop.ml

```diff
@@ -88,6 +89,26 @@ let unsupported_opt opt =
   in
   warning "unsupported option %s" opt
 
+let set_sat_solver s =
+  match s with
+  | Util.CDCL_Tableaux | CDCL -> O.set_sat_solver s
+  | _ when O.get_optimize () ->
+    warning
+      "Sat-solver %a is incompatible with optimization: ignoring command."
+      Util.pp_sat_solver
+      s
```

Comment:
This is not correct — the sat solver is stlil set on the Dolmen state even though it is not updated through options. Either the check needs to be in `handle_option` and `State.set sat_solver_key` must not be called, or we must call both `O.set_sat_solver` and `O.set_optimize false` here.

(Yes, the way we manage options is currently a mess, sorry)

Reply:
For further clarification — the SAT solver that is actually used by the binary is the one that is set on the Dolmen state. The options are only updated because of code in the SAT solvers that triggers different behavior depending on `Options.get_cdcl_tableaux` and related. We want to move away from code that depends on global `Options` values in the library as much as possible, but for now we have no choice but to update those options.